# Praktikum & Tugas - Data Understanding
**Kecerdasan Buatan**  
Nama: Mikhael Josua Roganda     
NIM: 42324023   
Prodi: D3 Teknologi Informasi 2024

## Bagian Praktikum

### 1. Import data menggunakan library pandas

In [1]:
import pandas as pd

path = "GlobalWeatherRepository.csv"
df = pd.read_csv(path)
df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


Dari output di atas, dapat diketahui beberapa karakteristik data seperti berikut ini:
1. Data terdiri dari 39.594 baris (indeks DataFrame dimulai dari 0 hingga 39.593)
2. Terdapat 41 kolom atau atribut dalam data


### 2. Melihat data awal dan akhir

In [2]:
df.head()

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.52,69.18,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.4,26.6,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.33,19.82,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.1,2.0,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.76,3.05,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.4,18.4,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.50,1.52,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.7,0.9,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.84,13.23,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.4,262.3,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55


In [3]:
df.tail()

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22
39593,Zimbabwe,Harare,-17.8178,31.0447,Africa/Harare,1733482800,12/6/2024 13:00,34.7,94.4,Sunny,...,12.950,13.505,1,2,5:13 AM,6:21 PM,9:43 AM,10:58 PM,Waxing Crescent,22


### 3. Mengambil data dari kolom temperature dan seterusnya

Jika ingin langsung menganalisis data dari kolom temperature, bisa langsung mengambil data temperature di kolom 7 dengan cara berikut.

In [4]:
df_noid = df.iloc[:, 7:]
df_noid

,temperature_celsius,temperature_fahrenheit,condition_text,wind_mph,wind_kph,wind_degree,wind_direction,pressure_mb,pressure_in,precip_mm,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,26.6,79.8,Partly Cloudy,8.3,13.3,338,NNW,1012,29.89,0.00,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,19.0,66.2,Partly cloudy,6.9,11.2,320,NW,1012,29.88,0.10,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,23.0,73.4,Sunny,9.4,15.1,280,W,1011,29.85,0.00,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,6.3,43.3,Light drizzle,7.4,11.9,215,SW,1007,29.75,0.30,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,26.0,78.8,Partly cloudy,8.1,13.0,150,SSE,1011,29.85,0.00,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,16.2,61.2,Sunny,4.0,6.5,158,SSE,1015,29.97,0.00,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,21.2,70.2,Partly cloudy,9.6,15.5,112,ESE,1013,29.91,0.15,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,22.2,71.9,Sunny,9.4,15.1,83,E,1016,30.00,0.00,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,36.0,96.8,Sunny,4.9,7.9,67,ENE,1007,29.75,0.00,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


### 4. Ringkasan statistik dengan describe()

Untuk memahami data dari ringkasan statistik, DataFrame menyediakan fungsi `describe()`.

In [5]:
df_noid.describe()

,temperature_celsius,temperature_fahrenheit,wind_mph,wind_kph,wind_degree,pressure_mb,pressure_in,precip_mm,precip_in,humidity,...,gust_kph,air_quality_Carbon_Monoxide,air_quality_Ozone,air_quality_Nitrogen_dioxide,air_quality_Sulphur_dioxide,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,moon_illumination
count,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,...,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000,39594.000000
mean,24.350599,75.832651,8.341135,13.427714,172.630853,1013.195914,29.919062,0.151742,0.005783,62.528237,...,19.520155,492.955139,62.538847,12.518648,8.974035,20.754174,39.177234,1.556574,2.260797,47.714856
std,8.130323,14.634566,10.624011,17.097136,101.820087,6.689236,0.197368,0.655896,0.025911,24.769931,...,18.625182,1020.366688,39.559135,24.459390,54.608971,45.741179,99.198453,0.891307,2.232790,35.038713
min,-21.900000,-7.400000,2.200000,3.600000,1.000000,971.000000,28.670000,0.000000,0.000000,2.000000,...,3.600000,-9999.000000,0.000000,0.000000,-9999.000000,0.185000,0.185000,1.000000,1.000000,0.000000
25%,20.200000,68.400000,4.300000,6.800000,88.000000,1010.000000,29.820000,0.000000,0.000000,44.000000,...,10.800000,212.000000,35.100000,0.740000,0.555000,3.800000,6.600000,1.000000,1.000000,12.000000
50%,26.100000,79.000000,7.200000,11.500000,168.000000,1013.000000,29.910000,0.000000,0.000000,67.000000,...,17.400000,293.700000,57.900000,2.600000,1.850000,10.360000,16.835000,1.000000,1.000000,47.000000
75%,29.200000,84.600000,11.600000,18.700000,258.000000,1016.000000,30.010000,0.030000,0.000000,83.000000,...,26.100000,443.347750,84.400000,11.500000,6.845000,22.940000,38.480000,2.000000,2.000000,82.000000
max,49.200000,120.600000,1841.200000,2963.200000,360.000000,1080.000000,31.890000,42.240000,1.660000,100.000000,...,2970.400000,38879.398000,480.700000,427.700000,400.340000,1614.100000,5554.810000,6.000000,10.000000,100.000000


- **count** menunjukkan jumlah total data atau observasi dalam kolom tersebut
- **mean** adalah nilai rata-rata aritmetik dari seluruh data pada kolom
- **std** (standar deviasi) mengukur seberapa besar variasi atau penyebaran data dari nilai rata-rata
- **min** menyatakan data terkecil pada kolom
- **25%** menunjukkan kuartil pertama (Q1) dari data pada kolom
- **50%** menunjukkan titik tengah data (median) atau kuartil kedua (Q2) dari data pada kolom
- **75%** menyatakan kuartil ketiga (Q3) dari data pada kolom
- **max** menyatakan data maksimum atau terbesar dari data pada kolom

### 5. Mengambil rata-rata kolom tertentu

Jika hanya ingin mengetahui data rata-rata tekanan udara dan kelembaban udara saja, bisa dengan menggunakan cara berikut ini.

In [6]:
df_noid[['pressure_mb','humidity']].mean()

pressure_mb    1013.195914
humidity         62.528237
dtype: float64

### 6. Korelasi antar data

Jika ingin mengetahui hubungan atau korelasi antar data, bisa dengan menggunakan fungsi `corr()`. Nilai korelasi berkisar dari -1 hingga 1.

- 1 = korelasi positif (ketika satu variabel naik, yang lain ikut naik)
- -1 = korelasi negatif (ketika satu variabel naik, yang lain turun)
- 0 = tidak ada korelasi

In [7]:
df_noid[['temperature_celsius', 'humidity','air_quality_Ozone']].corr()

,temperature_celsius,humidity,air_quality_Ozone
temperature_celsius,1.000000,-0.420825,0.334880
humidity,-0.420825,1.000000,-0.508499
air_quality_Ozone,0.334880,-0.508499,1.000000


Dari data di atas, dapat dipahami bahwa:

- **Temperature vs Air Quality Ozone**: korelasi positif, berarti semakin naik suhu maka kadar ozon juga meningkat.
- **Humidity vs Air Quality Ozone**: korelasi negatif, berarti semakin naik kelembaban udara maka kadar ozon menurun.
- **Kesimpulan**: kenaikan suhu meningkatkan kadar ozon, sedangkan kenaikan kelembaban udara menurunkan kadar ozon - pola ini sama seperti contoh yang diberikan pada modul, meskipun angka pastinya bisa sedikit berbeda tergantung data.

---
## Bagian 2 - Tugas

### Tugas 1
Ikuti langkah-langkah pada modul ini, kemudian analisa korelasi antara `temperature_fahrenheit`, `pressure_mb`, dan `air_quality_Sulphur_dioxide`.

Ringkasan statistik untuk ketiga kolom yang diminta:

In [8]:
df_noid[['temperature_fahrenheit', 'pressure_mb', 'air_quality_Sulphur_dioxide']].describe()

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
count,39594.000000,39594.000000,39594.000000
mean,75.832651,1013.195914,8.974035
std,14.634566,6.689236,54.608971
min,-7.400000,971.000000,-9999.000000
25%,68.400000,1010.000000,0.555000
50%,79.000000,1013.000000,1.850000
75%,84.600000,1016.000000,6.845000
max,120.600000,1080.000000,400.340000


Korelasi antar ketiga kolom tersebut:

In [9]:
df_noid[['temperature_fahrenheit', 'pressure_mb', 'air_quality_Sulphur_dioxide']].corr()

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
temperature_fahrenheit,1.000000,-0.486833,-0.023152
pressure_mb,-0.486833,1.000000,0.018620
air_quality_Sulphur_dioxide,-0.023152,0.018620,1.000000


**Interpretasi:**

- Temperature (Fahrenheit) vs Pressure (mb): korelasi bernilai negatif dan cukup terlihat (sekitar -0,49). Artinya, semakin tinggi suhu udara, tekanan udara cenderung sedikit menurun, dan sebaliknya.
- Temperature vs Sulphur Dioxide korelasinya sangat lemah dan mendekati 0 (sekitar -0,02), berarti hampir tidak ada hubungan linear antara suhu udara dengan kadar SO2.
- Pressure vs Sulphur Dioxide: korelasinya juga sangat lemah dan mendekati 0 (sekitar 0,02), sehingga tekanan udara tidak banyak berhubungan dengan kadar SO2.
- Catatan data pada kolom `air_quality_Sulphur_dioxide` ditemukan satu baris dengan nilai sentinel/anomali `-9999`, indikasi *missing value* yang tersembunyi di data mentah. Ini temuan penting pada tahap *Data Understanding* karena nilai tersebut sebaiknya dibersihkan (dihapus atau diimputasi) sebelum masuk ke tahap pemodelan, agar tidak mendistorsi hasil analisis statistik.

### Tugas 2
Gunakan dataset [Customer Shopping Trends Dataset](https://www.kaggle.com/datasets/iamsouravbanerjee/customer-shopping-trends-dataset).

- Analisa mean, std, min, max, Q1, Q2, Q3 untuk data `Previous Purchases`
- Analisa korelasi `Age` vs `Purchase Amount (USD)` dan `Purchase Amount (USD)` vs `Review Rating`

In [14]:
df2 = pd.read_csv("shopping_trends_updated.csv")
df2

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


#### Statistik deskriptif kolom `Previous Purchases`

In [15]:
df2['Previous Purchases'].describe()

count    3900.000000
mean       25.351538
std        14.447125
min         1.000000
25%        13.000000
50%        25.000000
75%        38.000000
max        50.000000
Name: Previous Purchases, dtype: float64

Nilai Q1, Q2 (median), dan Q3 secara eksplisit:

In [12]:
q1 = df2['Previous Purchases'].quantile(0.25)
q2 = df2['Previous Purchases'].quantile(0.50)
q3 = df2['Previous Purchases'].quantile(0.75)
print(f"Q1 = {q1}")
print(f"Q2 (median) = {q2}")
print(f"Q3 = {q3}")

Q1 = 13.0
Q2 (median) = 25.0
Q3 = 38.0


**Interpretasi:**

- **count** = 3900 data pelanggan.
- **mean** = 25,35, artinya rata-rata pelanggan sudah melakukan sekitar 25 kali pembelian sebelumnya.
- **std** = 14,45, menunjukkan variasi jumlah pembelian sebelumnya cukup besar antar pelanggan.
- **min** = 1 dan **max** = 50, rentang jumlah pembelian sebelumnya cukup lebar.
- **Q1** = 13, **Q2 (median)** = 25, **Q3** = 38, artinya separuh pelanggan memiliki riwayat pembelian sebelumnya antara 13 hingga 38 kali, dan distribusinya terlihat cukup merata/simetris karena median (25) hampir sama dengan mean (25,35).

#### Korelasi `Age` vs `Purchase Amount (USD)`

In [13]:
df2[['Age', 'Purchase Amount (USD)']].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


#### Korelasi `Purchase Amount (USD)` vs `Review Rating`

In [14]:
df2[['Purchase Amount (USD)', 'Review Rating']].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


**Interpretasi:**

- **Age vs Purchase Amount (USD)**: korelasi bernilai sekitar -0,01, sangat mendekati 0. Ini berarti usia pelanggan praktis tidak berhubungan dengan besar nilai pembelian yang mereka lakukan - pelanggan muda maupun tua cenderung membelanjakan jumlah yang serupa.
- **Purchase Amount (USD) vs Review Rating**: korelasi juga sangat lemah, sekitar 0,03. Artinya, besar kecilnya nominal pembelian tidak berhubungan secara linear dengan tinggi rendahnya rating ulasan yang diberikan pelanggan.

**Kesimpulan:** kedua pasangan variabel ini sama-sama menunjukkan hubungan linear yang sangat lemah (mendekati 0), sehingga `Age` dan `Review Rating` bukan prediktor yang kuat untuk `Purchase Amount (USD)` pada dataset ini.